[Joel Wong] iP

Jo3LW opened the pull request:
## Dukie

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Dukie frees your mind from having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use
- All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2223S1/ip/pull/342).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the ``` main``` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
add echo function

Commit:
Made it more neat. No code was added or removed

Commit:
added *.class to gitignore

Commit:
add ability to save user inputs

Commit:
add new Task class and allow Duke to mark and unmark tasks

Commit:
add Todo, Deadline and Event classes that inherit from Task. add functions to allow addition of events, todo and deadlines to the bot. Also cleared the code by creating methods for each action

Commit:
add test case to EXPECTED.TXT and INPUT.TXT under text-ui-test folder.

Commit:
add exception handling

Commit:
add delete function

Commit:
add comment to show delete method

Commit:
Add save function to Duke and some java coding standard for imports

Commit:
add date and time

Commit:
Merge tag 'Level-7'

# Conflicts:
#	src/main/java/Duke.java

Commit:
remove comments from the merger

Commit:
used OOP and create Parser class, Storage class, TaskList class and UI class

Commit:
delete unnecessary methods

Commit:
organise classes into packages

Commit:
Merge remote-tracking branch 'origin/add-gradle-support' into add-gradle-support

Commit:
add gradle support

Commit:
add Junit tests for Storage class

Commit:
package app as Jar file

Commit:
add JavaDocs

Commit:
edit code to fit coding standard

Commit:
add find function

Commit:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  add JavaDocs

Commit:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  edit code to fit coding standard

# Conflicts:
#	src/main/java/duke/Duke.java

Commit:
resolve conflict from merge

sherylkong18 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,183 @@
+import task.Todo;
+import task.TaskList;
+
+public class Parser {
```
Can add a description about what the Parser class does

sherylkong18 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,183 @@
+        return this.isAction;
+    }
+
+    private void bye(ArrayList<Task> listOfActions, File file) {
```
perhaps can give a more descriptive name to this method instead of bye (eg. a name related to what the method is doing)

sherylkong18 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,113 @@
+
+
+
+public class Storage {
```
brief description of what Storage class does

sherylkong18 commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,113 @@
+        Boolean d = !done.equals(" ");
+        //If td
+
+        if (t.equals("T")) {
```
can use switch statement for more concise code

sherylkong18 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,56 @@
+package task;
+
+public class Task {
```
brief description about what Task class does

sherylkong18 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     */
+    public void add(Task task) {
+        this.taskArray.add(task);
+        System.out.println("----------------------\n" + "Ok Solid you got this work to do:\n" +
```
printing out any texts to the user (all user interactions) can be handled by the UI class

sherylkong18 commented on src/main/java/ui/UI.java:
```diff
@@ -0,0 +1,65 @@
+package ui;
+
+public class UI {
```
brief description about what the UI class does

sherylkong18 reviewed (commented):
Overall looks good :)
Can improve on java docs and updating UI class

Commit:
add GUI using JavaFX

Commit:
delete unnecessary code in DukeException class

Commit:
Storage class: Code written to be more intuitive

Adding of tasks (e.g Todo, Event, Deadline) to the Tasklist have some
common behaviors (e.g. arr.addStart(), .mark()).

The common behaviors across person attribute classes cause code
duplication. Also the code written is not intuitive and messy.

By having methods with intuitive names to check what kind of tasks
it can make the code easier to understand.

By consolidating adding tasks that are marked or unmarked into the
Tasklist into 2 different methods, it can prevent code duplication
and make the code more intuitive.

Commit:
delete unused code in Duke.java

Commit:
add assertions to the class Bye, Delete, Parser

Commit:
improve code quality by changing lines in classes bye by removing arrow head code

Jo3LW committed:
Merge pull request #2 from Jo3LW/A-CodeQuality

Improve code quality

Jo3LW committed:
Merge branch 'master' into A-Assertions

Jo3LW committed:
Merge pull request #3 from Jo3LW/A-Assertions

Add assertions

Commit:
delete conflicts in merge

Commit:
add reminder function to inform users of any deadlines upcoming

Commit:
add JavaDocs to all classes

Commit:
edit code in lines 90 and 92 Storage.java that was wrong

Commit:
fixed bug in Parse class that did not allow Deadline object and Event object to be created due to outofindex error.

Jo3LW committed:
Merge pull request #5 from Jo3LW/B-Reminders

B reminders

Commit:
Merge commit '57cea28019a6cccd3420dcf86b0bd4c6f66f9adf'

* commit '57cea28019a6cccd3420dcf86b0bd4c6f66f9adf':
  fixed bug in Parse class that did not allow Deadline object and Event object to be created due to outofindex error.
  add JavaDocs to all classes
  add reminder function to inform users of any deadlines upcoming

# Conflicts:
#	src/main/java/storage/Storage.java

Commit:
replace wrong code in storage class

Commit:
handle DukeException except for Event class

Commit:
remove unnecessary System print lines and also add execute method in Duke to allow system to return Bye message and then exit

Commit:
fix bugs for Event class

Commit:
Merge branch 'master' into B-Reminders

* master:
  replace wrong code in storage class
  edit code in lines 90 and 92 Storage.java that was wrong

Commit:
remove system print message in storage class line 98

Commit:
improve GUI

Commit:
delete unneccssary code

Commit:
improve code quality

Jo3LW committed:
Update README.md

Jo3LW committed:
Update README.md

add link to download jar file.

Commit:
update ui

Commit:
update ui and improve code quality

Commit:
update Ui.png photo

Jo3LW committed:
Update README.md

change emoji as bowtie emoji does not show up on website

Jo3LW committed:
Update README.md

change emoji from smile to first place medal

Jo3LW committed:
Update README.md

add more to the aim at line 6
